### PR TITLE
Queue with stacks

### PR DESCRIPTION
**Pull request by AndyAgulue**

### CODE REVIEW

File: challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class PseudoQueue {
+    Stack stackOne;
+    Stack stackTwo;
+
+    public PseudoQueue(){
+        stackOne = new Stack();
+        stackTwo = new Stack();
+    }
+
+    public void push(Stack<Integer> topNode, int newValue) {
```
Comment (Mattpet26) [inline challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java:12]: topNode might be a misleading name for your stack.

File: challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package challenges.stacksandqueues;
+
+public class PseudoQueue {
```
Comment (Mattpet26) [inline challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java:3]: You didn't have to re-write your push and pop methods, pseudoQueue could just extend your Stack.

File: challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    public void deQueue(int value){
+       // int value;
+
+        // if both stacks are empty, throw an error
+        if(stackOne.isEmpty() && stackTwo.isEmpty()) {
+            System.out.println("Stack is empty");
+            System.exit(0);
+        }
+        // move elements from StackOne to stackTwo only if stack two is empty
```
Comment (Mattpet26) [inline challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java:37]: You are missing a bit of logic in the enqueue. You are checking below if stacktwo is empty, enqueue into stack one. But what if stackone is empty but stacktwo is not?

File: challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java (hunk 2 above)
Comment (AndyAgulue) [reply challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java:3]: Oh okay, Thank you

File: challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java (hunk 3 above)
Comment (AndyAgulue) [reply challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java:37]: Great point Matthew! It felt a little incomplete. thank you for the feedback

File: challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java (hunk 1 above)
Comment (AndyAgulue) [reply challenges/lib/src/main/java/challenges/stacksandqueues/PseudoQueue.java:12]: My naming convention needs work sometimes

**Pull request merged**